### PR TITLE
TypeError: Data must be a string or a buffer (#4072)

### PR DESCRIPTION
**Pull request by about-code**
**What kind of change does this PR introduce?**

Attempt to fix urgent issue #4072

**Did you add tests for your changes?**

<del>I'm sorry, no. Don't know how to reproduce the test case with unit tests. For the time being you may want to test it against [demo project by rguanghui](https://github.com/webpack/webpack/issues/4072#issuecomment-274257811) in #4072. Previous to the PR it should run into the Error described in the related issue. When trying to build it with the fixed version of `lib/dependencies/HarmonyExportImportedSpecifierDependency.js` it should print an error message.</del>

Added unit test which tests on the result of `HarmonyExportImportedSpecifierDependency#getHashCode(importedModule)` when method parameter is `null` or `undefined`.

**Does this PR introduce a breaking change?**
Don't think so.

### CODE REVIEW

File: lib/dependencies/HarmonyExportImportedSpecifierDependency.js [hunk 1]
```diff
@@ -146,7 +147,7 @@ class HarmonyExportImportedSpecifierDependency extends NullDependency {
 
 	getHashValue(importedModule) {
 		if(!importedModule) {
-			return importedModule;
+			return WebpackMissingModule.module(importedModule);
```
Comment (sokra) [inline lib/dependencies/HarmonyExportImportedSpecifierDependency.js]: An empty string will work fine here. `importedModule` is not set here.

Comment (about-code) [inline lib/dependencies/HarmonyExportImportedSpecifierDependency.js]: Here we go. Empty string did the trick as well. Didn't try that option so wasn't aware a missing module error will be issued elsewhere.

**Comment by sokra** [comment]
Thanks

**Comment by luckylooke** [comment]
How can I check from which webpack release is this merge included? 🤔

**Comment by sokra** [comment]
Open the commit, github shows the tags in which the commit is included.

**Comment by luckylooke** [comment]
@sokra can you please send me a screenshot? I could not found it :/ [Here is commit detail](http://prntscr.com/g13lj0) how it looks on my github, maybe you have enterprise github with more features?

**Comment by about-code** [comment]
@luckylooke By temporal reasoning you could infer that if this commit has been merged into master on Feb 28 its very likely that it's been included in the first release built from master after Feb 28 which was 2.3.0 on March 21st. If you like to convince yourself, click on the [merge commit](https://github.com/webpack/webpack/commit/0f255ca53bc0efdad24cf41754b9435fecaecef3) as suggested by @sokra. There's a blue header with a tag symbol :label:. If you expand the tags via `...` you see the complete history of tags including the commit. No surprise, it begins at 2.3.0.

**Comment by luckylooke** [comment]
@about-code thank you for your answer, I didn't get from @sokra answer that the merge commit is that one containing the info. Thank you both, now I moved forward.

**Pull request merged**